build.gradle

```diff
@@ -91,6 +91,13 @@ test.dependsOn('setupTools')
 if (!file("$generatedSrcDir").exists()) file("$generatedSrcDir").mkdirs()
 
 
+// Sources that contain templates to be generated.
+sourceSets {
+    templates {
+    }
+}
+sourceSets.templates.resources.srcDir 'src/common'
```

Comment:
I guess a little explanation is in order here...  There were a couple of classes that changed class signatures between the versions 2018.1->.2, and again 2018.2->.3.  These incompatibilities require separate sources.

Rather than going the route of separate trees for the versions like we used to have, I opted to "generate" the proper sources from templates that are placed in the source tree where their original files should have occurred.  Then, at build time, we copy them into the /gen/... tree and rename them to *.java files.  Most of the changes in the build script are to accommodate that.  I've made detection of them automatic, too, so we don't have to specify the files or the correct versions in any configuration files.  The file extensions themselves have the version information, and the closest numbered file to the current version without being larger is the one that will be used.

All of the new template files (three sets) are the minimal set of changes that I could get away with.  In most cases, I just created a new base class in the templates and changed the inheritance of the older (now incompatible) classes to use the new base classes.  This way, I minimized the source changes in the original files.